app/services/staff_action_logger.rb

```diff
@@ -82,10 +82,11 @@ def log_post_deletion(deleted_post, opts = {})
       "post_number: #{deleted_post.post_number}",
       "raw: #{truncate(deleted_post.raw)}",
     ]
+    details = [] if opts[:permanent]
```

Comment:
Perhaps instead of populating all details and then resetting them, it would be clearer if we do an if-else here. Similar in `#log_topic_delete_recover`. WDYT?